astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
+                        len(other_class) != 1+len(self.differentials)):
+                    raise ValueError("If multiple classes are passed in to "
+                                     "change the representations of attached "
+                                     "differentials, the number of classes must "
+                                     "be equal to 1 + the number of "
+                                     "differentials. (got {0})"
+                                     .format(len(other_class)))
+
+                new_rep = other_class[0].from_cartesian(self.to_cartesian())
+
+                new_diffs = []
+                for i,diff in enumerate(self.differentials):
+                    new_diffs.append(diff.represent_as(other_class[1+i],
+                                                       base=self))
+                new_rep._differentials = tuple(new_diffs)
+
+            else:
+                new_rep = other_class.from_cartesian(self.to_cartesian())
+
+            return new_rep
+
+    def with_differentials(self, differentials):
+        """
+        Create a new representation with the same positions as this
+        representation, but with new differentials.
+
+        Parameters
+        ----------
+        differentials : Sequence of `~astropy.coordinates.BaseDifferential`
+            The differentials for the new representation to have.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but with the ``differentials`` as
+            its differentials.
+        """
+        # Implemented this way for (1) performance (skips going through the
+        # initializer), and (2) will preserve any additional attributes a user
+        # may have set
+
+        new_differentials = self._validate_differentials(differentials)
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = new_differentials
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def without_differentials(self):
+        """
+        Create a new representation with the same positions as this
+        representation, but without any differentials.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but without the ``differentials``.
+        """
+
+        # Again, implemented this way for (1) performance (skips going through
+        # the initializer), and (2) will preserve any additional attributes a
+        # user may have set
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
```

Comment:
But also: just return `self` if the representation has no differentials; then,
`r1.without_differentials().to_cartesian()` is very fast for that case.